project-api/ballerina-projects/src/main/java/io/ballerina/projects/writers/BaloWriter.java

```diff
@@ -29,18 +59,347 @@
+        }
+
+        // Add other module docs
+        File modulesSourceDir = new File(String.valueOf(packageSourceDir.resolve(MODULES)));
+        File[] directoryListing = modulesSourceDir.listFiles();
+
+        if (directoryListing != null) {
+            for (File moduleDir : directoryListing) {
+                if (moduleDir.isDirectory()) {
+                    // Get `Module.md` path
+                    Path otherModuleMd = packageSourceDir.resolve(MODULES).resolve(moduleDir.getName())
+                            .resolve(moduleMdFileName);
+                    // Create `package.module` folder, if `Module.md` path exists
+                    if (otherModuleMd.toFile().exists()) {
+                        Path otherModuleDirInBaloDocs = modulesDirInBaloDocs
+                                .resolve(pkgName + "." + moduleDir.getName());
+                        Files.createDirectory(otherModuleDirInBaloDocs);
+                        Path otherModuleMdInBaloDocs = modulesDirInBaloDocs
+                                .resolve(pkgName + "." + moduleDir.getName()).resolve(moduleMdFileName);
+                        Files.copy(otherModuleMd, otherModuleMdInBaloDocs);
+                    }
+                }
+            }
+        }
+    }
+
+    private static void addPackageSource(Path root, Path packageSourceDir, String defaultPackageName)
+            throws IOException {
+        // create the module directory in zip
+        Path packageInBalo = root.resolve(MODULES);
+        Files.createDirectory(packageInBalo);
+
+        // add default module
+        Path defaultPkgDirInBalo = packageInBalo.resolve(defaultPackageName);
+        Files.createDirectory(defaultPkgDirInBalo);
+
+        // copy resources directory
+        copyResourcesDir(packageSourceDir, defaultPkgDirInBalo);
+        // only add .bal files in the package root
+        copyBallerinaSrcFiles(packageSourceDir, defaultPkgDirInBalo);
+
+        // add other modules
+        File modulesSourceDir = new File(String.valueOf(packageSourceDir.resolve(MODULES)));
+        File[] directoryListing = modulesSourceDir.listFiles();
+
+        if (directoryListing != null) {
+            for (File moduleDir : directoryListing) {
+                if (moduleDir.isDirectory()) {
+                    // add module
+                    Path moduleDirInBalo = packageInBalo.resolve(defaultPackageName + "." + moduleDir.getName());
+                    Files.createDirectory(moduleDirInBalo);
+
+                    // copy resources directory
+                    copyResourcesDir(moduleDir.toPath(), moduleDirInBalo);
+                    // only add .bal files in the module root
+                    copyBallerinaSrcFiles(moduleDir.toPath(), moduleDirInBalo);
+                }
+            }
+        }
+    }
+
+    private static void copyResourcesDir(Path sourceDir, Path sourceDirInBalo) throws IOException {
+        Path resourcesDir = sourceDirInBalo.resolve("resources");
```

Comment:
We can use `ProjectConstants.RESOURCE_DIR_NAME` instead of the hardcoded string `"resources"`

Reply:
Added